canopen/pdo.py

```diff
@@ -116,14 +116,18 @@ def stop(self):
 class Maps(collections.Mapping):
     """A collection of transmit or receive maps."""
 
-    def __init__(self, com_offset, map_offset, pdo_node):
+    def __init__(self, com_offset, map_offset, pdo_node, cob_base=None):
         self.maps = {}
         for map_no in range(128):
             if com_offset + map_no in pdo_node.node.object_dictionary:
-                self.maps[map_no + 1] = Map(
+                new_map = Map(
                     pdo_node,
                     pdo_node.node.sdo[com_offset + map_no],
                     pdo_node.node.sdo[map_offset + map_no])
+                # Generate default COB-IDs for predefined connection set
+                if cob_base is not None and map_no < 4:
+                    new_map.predefined_cob_id = cob_base + map_no * 0x100 + pdo_node.node.id
```

Comment:
I think you can set `cob_id` directly instead of having a separate attribute.

Reply:
That would break the logic in `read()` and `save()`, which will use the device's configured COB-ID from the record only if the attribute was still `None`.

I intentionally left this as a conscious decision for the API user, because the nodes need not be configured to use the predefined connection set.  The PDO configuration record's COB-ID entry may be set to anything really, especially if some nodes should cross-communicate, setting one's RxPDO COB to the other's TxPDO COB for example.